conjure-java-core/src/integrationInput/java/com/palantir/product/ErrorServiceAsync.java

```diff
@@ -0,0 +1,386 @@
+                super(errorCode, TestErrors.NOT_FOUND.name(), errorInstanceId, parameters);
+            }
+        }
+
+        final class DifferentNamespace extends ConjureErrors.BaseEndpointError<ConjureErrors.EmptyErrorParameters>
+                implements TestMultipleErrorsAndPackagesResponse {
+            @JsonCreator
+            DifferentNamespace(
+                    @JsonProperty("errorCode") String errorCode,
+                    @JsonProperty("errorInstanceId") String errorInstanceId) {
+                super(
+                        errorCode,
+                        EndpointSpecificTwoErrors.DIFFERENT_NAMESPACE.name(),
+                        errorInstanceId,
+                        ConjureErrors.EMPTY_ERROR_PARAMETERS_INSTANCE);
+            }
+        }
+
+        final class DifferentPackage extends ConjureErrors.BaseEndpointError<ConjureErrors.EmptyErrorParameters>
+                implements TestMultipleErrorsAndPackagesResponse {
+            @JsonCreator
+            DifferentPackage(
+                    @JsonProperty("errorCode") String errorCode,
+                    @JsonProperty("errorInstanceId") String errorInstanceId) {
+                super(
+                        errorCode,
+                        com.palantir.another.EndpointSpecificErrors.DIFFERENT_PACKAGE.name(),
+                        errorInstanceId,
+                        ConjureErrors.EMPTY_ERROR_PARAMETERS_INSTANCE);
+            }
+        }
+    }
+
+    sealed interface TestEmptyBodyResponse
+            permits TestEmptyBodyResponse.Success, TestEmptyBodyResponse.InvalidArgument {
+        record Success() implements TestEmptyBodyResponse {
+            @JsonCreator
+            public static Success create() {
+                return new Success();
+            }
+        }
+
+        final class InvalidArgument extends ConjureErrors.BaseEndpointError<TestErrors.InvalidArgumentParameters>
+                implements TestEmptyBodyResponse {
+            @JsonCreator
+            InvalidArgument(
+                    @JsonProperty("errorCode") String errorCode,
+                    @JsonProperty("errorInstanceId") String errorInstanceId,
+                    @JsonProperty("parameters") TestErrors.InvalidArgumentParameters parameters) {
+                super(errorCode, TestErrors.INVALID_ARGUMENT.name(), errorInstanceId, parameters);
+            }
+        }
+    }
+
+    sealed interface TestBinaryResponse permits TestBinaryResponse.Success, TestBinaryResponse.InvalidArgument {
+        record Success(@MustBeClosed InputStream value) implements TestBinaryResponse {
+            public Success {
+                Preconditions.checkArgumentNotNull(value, "value cannot be null");
+            }
+        }
+
+        final class InvalidArgument extends ConjureErrors.BaseEndpointError<TestErrors.InvalidArgumentParameters>
+                implements TestBinaryResponse {
+            @JsonCreator
```

Comment:
Let's make the annotation as specific as possible, which reduces the risk of jackson changes either breaking deserialization, or inadvertently causing us to successfully deserialize shapes we don't expect:
```suggestion
            @JsonCreator(Mode.PROPERTIES)
```